contrib/ci/dependencies.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2017-2018 Dell, Inc.
+# Copyright (C) 2020 Intel, Inc.
+#
+# SPDX-License-Identifier: LGPL-2.1+
+#
+import os
+import sys
+import xml.etree.ElementTree as etree
+
+def parse_dependencies(OS, SUBOS, requested_type):
+    deps = []
+    dep = ''
+    directory = os.path.dirname(sys.argv[0])
+    tree = etree.parse(os.path.join(directory, "dependencies.xml"))
+    root = tree.getroot()
+    for child in root:
+        if not "type" in child.attrib or not "id" in child.attrib:
+            continue
+        for distro in child:
+            if not "id" in distro.attrib:
+                continue
+            if distro.attrib["id"] != OS:
+                continue
+            packages = distro.findall("package")
+            for package in packages:
+                if SUBOS:
+                    if not 'variant' in package.attrib:
+                        continue
+                    if package.attrib['variant'] != SUBOS:
+                        continue
+                if package.text:
+                    dep = package.text
+                else:
+                    dep = child.attrib["id"]
+                if child.attrib["type"] == requested_type and dep:
+                    deps.append(dep)
+    return deps
+
+if __name__ == '__main__':
+
+    try:
+        import distro
+        target = distro.linux_distribution()[0]
+    except ModuleNotFoundError:
+        target = None
+
+    target = os.getenv('OS', target)
+    if target is None:
+        print("Missing OS environment variable")
+        sys.exit(1)
```

Comment:
The main reason for using OS in the original script was that it was a very convenient way for Travis to pass to the job from the environment what it needs to be.

For the purpose of a standalone script I would think supporting argument parsing would be more convenient (and you can have a --help command too then)

Reply:
I can do that, the question if it is desirable to retain the env variable, too.

Reply:
Yeah I think retain it so that it can still work with the docker build script